swent-epfl-groupe07/eduverse-android
# Feature/display and save video

## Overview
This PR introduces the video storage and display features for EduVerse, enhancing the media experience within the app by enabling users to save, post, and view videos with a vertical scrolling interface.

## Key Features
- Video storage functionality to securely save user-generated content
- Save and post functionalities for both photos and videos
- Vertical scrolling in VideoScreen for seamless navigation through media
- MVVM architecture for organized publication management
- Firebase Storage integration for media files (photos and videos)
- Full test coverage for UI interactions and functionalities

##Technical Implementation
- Architecture: MVVM pattern ensuring clean separation of concerns
- State Management: Kotlin Flow for reactive updates and seamless UI transitions
- Database: Firebase Firestore for managing publication data
- Storage: Firebase Storage for both photos and videos
- UI: Jetpack Compose with Material3 for consistent design
Testing: Comprehensive UI and functional tests with Android test frameworks and mocked Firebase dependencies

## Testing
- UI tests covering all major interactions, including video scrolling, saving, and posting
- Tests for handling loading, success, and error states
- Mocked Firebase instances to simulate data interactions
- Achieved test coverage > ?

## Screenshots
Here is a preview of the VideoScreen with vertical scrolling:
<img width="222" alt="tiktokWhileScrolling" src="https://github.com/user-attachments/assets/2767c465-59e2-4dd7-9514-61934071040e">


## Related Issues
- Implements video storage and publication features
- Addresses user needs for media display and sharing
- Testing Instructions
- Run the app and navigate to the VideoScreen.
- Test the save functionality for both photos and videos.
- Scroll through videos to confirm smooth vertical navigation.
- Check loading and error handling states during posting and viewing. 

## Notes
- Requires Firebase setup for Storage and Firestore
- Ensure appropriate indexing in Firestore for optimal performance
